src/tracer/tracer.ts

```diff
@@ -16,6 +24,7 @@ interface TracerArgs {
 export class AutoblocksTracer {
   private _traceId: string | undefined;
   private properties: ArbitraryProperties;
+  private evaluatorSemaphoreRegistry: Record<string, Semaphore> = {}; // Evaluator id -> Semaphore
```

Comment:
This shouldn't be attached to the tracer instance, it needs to be global. The concurrency setting on an evaluator is a global setting, not something that's enforced per tracer instance